martha/martha_v3.js

```diff
@@ -398,31 +424,66 @@ async function marthaV3Handler(req, res) {
         }
     }
 
+    // TODO: should we know from the DrsType whether an `access_id` is expected
+    // -or- should we use the absence of an `access_id` to know we don't need a second DRS server request
+
     let drsResponse;
+    let accessId;
     try {
         drsResponse = responseParser(response);
+        accessId = getDrsAccessId(drsResponse, accessMethodType);
     } catch (error) {
         logAndSendServerError(res, error, 'Received error while parsing response from DRS URL.');
         return;
     }
 
+    // do more stuff here, i.e. get signed URL
+    // first, fetch access token from bond
+    // TODO: allow this to be not done based on requested data
+    let accessToken;
+    if (bondAccessTokenUrl && accessId) {
```

Comment:
1. Replace the comments with something more descriptive
2. Don't access this `if` if the client hasn't requested the `accessUrl`